server/config/config.go

```diff
@@ -18,6 +18,7 @@ type StoredConfig struct {
 // Config represents the the metadata handed to all request runners (command,
 // http).
 type Config struct {
+	Provider               string
```

Comment:
Same here, I think `CalendarProviderName` would be a good name for this. `Name` may be overkill though. I think we'll want some sort of struct like `CalendarProviderMetadata` to store bot username info etc.